hphp/runtime/vm/jit/relocation-arm.cpp

```diff
@@ -209,10 +217,8 @@ bool relocateSmashable(Env& env, Instruction* src, Instruction* dest,
   }
   env.updateInternalRefs = true;
   FTRACE(3,
-         "Relocated smashable at src 0x{:08x} ",
-         "with target 0x{:08x} to 0x{:08x} (0x{:08x})\n",
-         (uint64_t)srcAddr, (uint64_t)target,
-         *((uint32_t*)destAddr), (uint64_t)destAddr);
+         "Relocated and optimized smashable at src {} with target {} to {}.\n",
+         srcAddrActual, target, env.destBlock.toDestAddress(destAddr));
```

Comment:
You might want to print the virtual source too, the metadata won't reference the "real" source.